File: app/src/main/java/com/example/firefighters/viewmodels/UnitViewModel.java
package com.example.firefighters.viewmodels;

import com.example.firefighters.models.EmergencyModel;
import com.example.firefighters.models.UnitModel;
import com.example.firefighters.repositories.EmergencyRepository;
import com.example.firefighters.repositories.UnitRepository;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.Query;
import com.google.firebase.firestore.QuerySnapshot;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

public class UnitViewModel extends ViewModel {

    private UnitRepository repository;

    public void init() {
        if (repository != null)
            return;
        repository = UnitRepository.getInstance();
    }
    public LiveData<UnitModel> getUnitModel(String unitName) {
        return repository.getUnitModel(unitName);
    }
    public LiveData<QuerySnapshot> getUnitsQuery() {
        return repository.getUnitsQuery();
    }

    public LiveData<Integer> saveUnit(UnitModel unitModel) {
        return repository.saveUnit(unitModel);
    }
}

File: app/src/main/java/com/example/firefighters/viewmodels/models/UnitViewModel.kt
package com.example.firefighters.viewmodels.models

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.firefighters.models.UnitModel
import com.example.firefighters.repositories.UnitRepository
import com.google.firebase.firestore.QuerySnapshot

class UnitViewModel(private val mRepository: UnitRepository) : ViewModel() {

    fun getUnitModel(unitName: String?): LiveData<UnitModel?> {
        return mRepository.getUnitModel(unitName)
    }

    val unitsQuery: LiveData<QuerySnapshot>
        get() = mRepository.unitsQuery

    fun saveUnit(unitModel: UnitModel?): LiveData<Int> {
        return mRepository.saveUnit(unitModel!!)
    }

    class Factory(
        private val repository: UnitRepository
    ) : ViewModelProvider.NewInstanceFactory() {
        override fun <T : ViewModel> create(modelClass: Class<T>): T {
            return UnitViewModel(repository) as T
        }
    }
}
